\documentclass{article}\usepackage[]{graphicx}\usepackage[]{color}
%% maxwidth is the original width if it is less than linewidth
%% otherwise use linewidth (to make sure the graphics do not exceed the margin)
\makeatletter
\def\maxwidth{ %
  \ifdim\Gin@nat@width>\linewidth
    \linewidth
  \else
    \Gin@nat@width
  \fi
}
\makeatother

\definecolor{fgcolor}{rgb}{0.345, 0.345, 0.345}
\newcommand{\hlnum}[1]{\textcolor[rgb]{0.686,0.059,0.569}{#1}}%
\newcommand{\hlstr}[1]{\textcolor[rgb]{0.192,0.494,0.8}{#1}}%
\newcommand{\hlcom}[1]{\textcolor[rgb]{0.678,0.584,0.686}{\textit{#1}}}%
\newcommand{\hlopt}[1]{\textcolor[rgb]{0,0,0}{#1}}%
\newcommand{\hlstd}[1]{\textcolor[rgb]{0.345,0.345,0.345}{#1}}%
\newcommand{\hlkwa}[1]{\textcolor[rgb]{0.161,0.373,0.58}{\textbf{#1}}}%
\newcommand{\hlkwb}[1]{\textcolor[rgb]{0.69,0.353,0.396}{#1}}%
\newcommand{\hlkwc}[1]{\textcolor[rgb]{0.333,0.667,0.333}{#1}}%
\newcommand{\hlkwd}[1]{\textcolor[rgb]{0.737,0.353,0.396}{\textbf{#1}}}%
\let\hlipl\hlkwb

\usepackage{framed}
\makeatletter
\newenvironment{kframe}{%
 \def\at@end@of@kframe{}%
 \ifinner\ifhmode%
  \def\at@end@of@kframe{\end{minipage}}%
  \begin{minipage}{\columnwidth}%
 \fi\fi%
 \def\FrameCommand##1{\hskip\@totalleftmargin \hskip-\fboxsep
 \colorbox{shadecolor}{##1}\hskip-\fboxsep
     % There is no \\@totalrightmargin, so:
     \hskip-\linewidth \hskip-\@totalleftmargin \hskip\columnwidth}%
 \MakeFramed {\advance\hsize-\width
   \@totalleftmargin\z@ \linewidth\hsize
   \@setminipage}}%
 {\par\unskip\endMakeFramed%
 \at@end@of@kframe}
\makeatother

\definecolor{shadecolor}{rgb}{.97, .97, .97}
\definecolor{messagecolor}{rgb}{0, 0, 0}
\definecolor{warningcolor}{rgb}{1, 0, 1}
\definecolor{errorcolor}{rgb}{1, 0, 0}
\newenvironment{knitrout}{}{} % an empty environment to be redefined in TeX

\usepackage{alltt}
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
\begin{document}

\begin{knitrout}
\definecolor{shadecolor}{rgb}{0.969, 0.969, 0.969}\color{fgcolor}\begin{kframe}
\begin{alltt}
\hlcom{#source("https://bioconductor.org/biocLite.R")}
\hlcom{#biocLite("GenomicRanges")}
\hlcom{#biocLite("rtracklayer")}
\hlcom{#biocLite("GenomicAlignments")}
\hlkwd{library}\hlstd{(GenomicRanges)}
\end{alltt}


{\ttfamily\noindent\itshape\color{messagecolor}{\#\# Loading required package: stats4}}

{\ttfamily\noindent\itshape\color{messagecolor}{\#\# Loading required package: BiocGenerics}}

{\ttfamily\noindent\itshape\color{messagecolor}{\#\# Loading required package: parallel}}

{\ttfamily\noindent\itshape\color{messagecolor}{\#\# \\\#\# Attaching package: 'BiocGenerics'}}

{\ttfamily\noindent\itshape\color{messagecolor}{\#\# The following objects are masked from 'package:parallel':\\\#\# \\\#\#\ \ \ \  clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,\\\#\#\ \ \ \  clusterExport, clusterMap, parApply, parCapply, parLapply,\\\#\#\ \ \ \  parLapplyLB, parRapply, parSapply, parSapplyLB}}

{\ttfamily\noindent\itshape\color{messagecolor}{\#\# The following objects are masked from 'package:stats':\\\#\# \\\#\#\ \ \ \  IQR, mad, sd, var, xtabs}}

{\ttfamily\noindent\itshape\color{messagecolor}{\#\# The following objects are masked from 'package:base':\\\#\# \\\#\#\ \ \ \  anyDuplicated, append, as.data.frame, cbind, colMeans,\\\#\#\ \ \ \  colnames, colSums, do.call, duplicated, eval, evalq, Filter,\\\#\#\ \ \ \  Find, get, grep, grepl, intersect, is.unsorted, lapply,\\\#\#\ \ \ \  lengths, Map, mapply, match, mget, order, paste, pmax,\\\#\#\ \ \ \  pmax.int, pmin, pmin.int, Position, rank, rbind, Reduce,\\\#\#\ \ \ \  rowMeans, rownames, rowSums, sapply, setdiff, sort, table,\\\#\#\ \ \ \  tapply, union, unique, unsplit, which, which.max, which.min}}

{\ttfamily\noindent\itshape\color{messagecolor}{\#\# Loading required package: S4Vectors}}

{\ttfamily\noindent\itshape\color{messagecolor}{\#\# \\\#\# Attaching package: 'S4Vectors'}}

{\ttfamily\noindent\itshape\color{messagecolor}{\#\# The following object is masked from 'package:base':\\\#\# \\\#\#\ \ \ \  expand.grid}}

{\ttfamily\noindent\itshape\color{messagecolor}{\#\# Loading required package: IRanges}}

{\ttfamily\noindent\itshape\color{messagecolor}{\#\# Loading required package: GenomeInfoDb}}\begin{alltt}
\hlkwd{library}\hlstd{(GenomicAlignments)}
\end{alltt}


{\ttfamily\noindent\itshape\color{messagecolor}{\#\# Loading required package: SummarizedExperiment}}

{\ttfamily\noindent\itshape\color{messagecolor}{\#\# Loading required package: Biobase}}

{\ttfamily\noindent\itshape\color{messagecolor}{\#\# Welcome to Bioconductor\\\#\# \\\#\#\ \ \ \  Vignettes contain introductory material; view with\\\#\#\ \ \ \  'browseVignettes()'. To cite Bioconductor, see\\\#\#\ \ \ \  'citation("{}Biobase"{})', and for packages 'citation("{}pkgname"{})'.}}

{\ttfamily\noindent\itshape\color{messagecolor}{\#\# Loading required package: DelayedArray}}

{\ttfamily\noindent\itshape\color{messagecolor}{\#\# Loading required package: matrixStats}}

{\ttfamily\noindent\itshape\color{messagecolor}{\#\# \\\#\# Attaching package: 'matrixStats'}}

{\ttfamily\noindent\itshape\color{messagecolor}{\#\# The following objects are masked from 'package:Biobase':\\\#\# \\\#\#\ \ \ \  anyMissing, rowMedians}}

{\ttfamily\noindent\itshape\color{messagecolor}{\#\# \\\#\# Attaching package: 'DelayedArray'}}

{\ttfamily\noindent\itshape\color{messagecolor}{\#\# The following objects are masked from 'package:matrixStats':\\\#\# \\\#\#\ \ \ \  colMaxs, colMins, colRanges, rowMaxs, rowMins, rowRanges}}

{\ttfamily\noindent\itshape\color{messagecolor}{\#\# The following object is masked from 'package:base':\\\#\# \\\#\#\ \ \ \  apply}}

{\ttfamily\noindent\itshape\color{messagecolor}{\#\# Loading required package: Biostrings}}

{\ttfamily\noindent\itshape\color{messagecolor}{\#\# Loading required package: XVector}}

{\ttfamily\noindent\itshape\color{messagecolor}{\#\# \\\#\# Attaching package: 'Biostrings'}}

{\ttfamily\noindent\itshape\color{messagecolor}{\#\# The following object is masked from 'package:DelayedArray':\\\#\# \\\#\#\ \ \ \  type}}

{\ttfamily\noindent\itshape\color{messagecolor}{\#\# The following object is masked from 'package:base':\\\#\# \\\#\#\ \ \ \  strsplit}}

{\ttfamily\noindent\itshape\color{messagecolor}{\#\# Loading required package: Rsamtools}}\begin{alltt}
\hlkwd{library}\hlstd{(rtracklayer)}
\hlkwd{library}\hlstd{(parallel)}
\hlkwd{source}\hlstd{(}\hlkwc{file} \hlstd{=} \hlstr{"~/ISGlobal/Scripts/Funcions_1.R"}\hlstd{)}
\end{alltt}
\end{kframe}
\end{knitrout}


\begin{knitrout}
\definecolor{shadecolor}{rgb}{0.969, 0.969, 0.969}\color{fgcolor}\begin{kframe}
\begin{alltt}
\hlstd{in_path} \hlkwb{<-} \hlstr{"/home/lucas/ISGlobal/Chip_Seq/DATA/Aligns"}
\hlstd{files} \hlkwb{<-} \hlkwd{list.files}\hlstd{(}\hlkwc{path} \hlstd{=} \hlstr{"/home/lucas/ISGlobal/Chip_Seq/DATA/Aligns"}\hlstd{)}
\hlstd{bams} \hlkwb{<-} \hlstd{files[}\hlkwd{substrR}\hlstd{(files,} \hlnum{8}\hlstd{)} \hlopt{==} \hlstr{"sort.bam"}\hlstd{]}
\hlstd{bams} \hlkwb{<-} \hlkwd{lapply}\hlstd{(bams,} \hlkwa{function}\hlstd{(}\hlkwc{x}\hlstd{)} \hlkwd{paste0}\hlstd{(in_path,}\hlstr{"/"}\hlstd{,x))}
\hlstd{me_bams} \hlkwb{<-} \hlstd{bams[}\hlkwd{substrR}\hlstd{(bams,} \hlnum{11}\hlstd{)} \hlopt{==} \hlstr{"me_sort.bam"}\hlstd{]}
\hlstd{aligns} \hlkwb{<-} \hlkwd{lapply}\hlstd{(me_bams, readGAlignmentPairs)}
\end{alltt}
\end{kframe}
\end{knitrout}

%% ---------------------------------------------------------------------------
\subsection{Calculating Coverage}
\begin{knitrout}
\definecolor{shadecolor}{rgb}{0.969, 0.969, 0.969}\color{fgcolor}\begin{kframe}
\begin{alltt}
\hlstd{cov} \hlkwb{<-} \hlkwd{lapply}\hlstd{(aligns, coverage)}
\end{alltt}
\end{kframe}
\end{knitrout}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section{Normalization}%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%% ----------------------------------------------------------------------------
\subsection{By Number of Mapped Reads} %% -------------------------------------
%% samtools view -F 0x4 foo_sorted.bam | cut -f 1 | sort | uniq | wc -l 

\noindent To get the number of mapped reads in the sample we use \texttt{samtools} as follows:

\begin{verbatim}
samtools view -F 0x4 foo.sorted.bam | cut -f 1 | sort | uniq | wc -l
-F : tells samtools to exclude given flag. Flag 0x4 means unaligned.
\end{verbatim}

\begin{knitrout}
\definecolor{shadecolor}{rgb}{0.969, 0.969, 0.969}\color{fgcolor}\begin{kframe}
\begin{alltt}
\hlstd{nreads} \hlkwb{<-} \hlkwd{c}\hlstd{()}
\hlkwa{for} \hlstd{(file} \hlkwa{in} \hlstd{me_bams) \{}
  \hlstd{cmd} \hlkwb{<-} \hlkwd{paste0}\hlstd{(}\hlstr{"/home/lucas/Programs/samtools-1.3.1/samtools view -F 0x4 "}\hlstd{, file,} \hlstr{" | cut -f 1 | sort | uniq | wc -l"}\hlstd{)}
  \hlstd{nreads} \hlkwb{<-} \hlkwd{c}\hlstd{(nreads,}\hlkwd{system}\hlstd{(cmd,} \hlkwc{intern} \hlstd{=} \hlnum{TRUE}\hlstd{))}
\hlstd{\}}
\hlkwd{names}\hlstd{(nreads)} \hlkwb{<-} \hlstd{me_bams}
\end{alltt}
\end{kframe}
\end{knitrout}

\subsection{Normalize Coverage}
\noindent Normalize the coverage divding by the number of mapped reads (abd multiplying by a huge factor).
\begin{knitrout}
\definecolor{shadecolor}{rgb}{0.969, 0.969, 0.969}\color{fgcolor}\begin{kframe}
\begin{alltt}
\hlstd{norm_cov} \hlkwb{<-} \hlkwd{c}\hlstd{()}
\hlkwa{for} \hlstd{(i} \hlkwa{in} \hlkwd{range}\hlstd{(}\hlnum{1}\hlstd{,}\hlkwd{length}\hlstd{(nreads)))\{}
  \hlstd{norm_cov} \hlkwb{<-} \hlkwd{c}\hlstd{(norm_cov, cov[[i]]}\hlopt{/}\hlkwd{as.numeric}\hlstd{(nreads[i])}\hlopt{*}\hlnum{1000000000}\hlstd{)}
\hlstd{\}}
\hlkwd{names}\hlstd{(norm_cov)} \hlkwb{<-} \hlstd{me_bams}
\end{alltt}


{\ttfamily\noindent\bfseries\color{errorcolor}{\#\# Error in names(norm\_cov) <- me\_bams: 'names' attribute [5] must be the same length as the vector [2]}}\end{kframe}
\end{knitrout}


\begin{knitrout}
\definecolor{shadecolor}{rgb}{0.969, 0.969, 0.969}\color{fgcolor}\begin{kframe}
\begin{alltt}
\hlkwa{for} \hlstd{(i} \hlkwa{in} \hlkwd{range}\hlstd{(}\hlnum{1}\hlstd{,} \hlkwd{length}\hlstd{(norm_cov)))\{}
  \hlstd{con} \hlkwb{<-} \hlkwd{gsub}\hlstd{(}\hlstr{"_sort.bam"}\hlstd{,} \hlstr{"_cov.bed"}\hlstd{, bams[i])}
  \hlkwd{export}\hlstd{(norm_cov[[i]],} \hlkwc{con} \hlstd{= con,} \hlkwc{format} \hlstd{=} \hlstr{"BED"}\hlstd{)}
\hlstd{\}}
\end{alltt}
\end{kframe}
\end{knitrout}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section{Substractions}%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{knitrout}
\definecolor{shadecolor}{rgb}{0.969, 0.969, 0.969}\color{fgcolor}\begin{kframe}
\begin{alltt}
subs_list <- \hlkwd{combn}(me_bams, 2, simplify = FALSE)
\hlkwd{for} (element in subs_list)\{
  substract <- 
\}

substraction_A7K9_E5K9 <- norm_cov[[1]] - norm_cov[[2]]
\hlkwd{export}(substraction_A7K9_E5K9, con = \hlstr{"/home/lucas/ISGlobal/Chip_Seq/DATA/Substractions/A7K9_E5K9_substr.bed"}, format = \hlstr{"BED"})

smooth_sub <- \hlkwd{runmean}(substraction_A7K9_E5K9, 10, endrule = \hlstr{"constant"})
\hlkwd{export}(smooth_sub, con = \hlstr{"/home/lucas/ISGlobal/Chip_Seq/DATA/Substractions/A7K9_E5K9_substr_smooth.bed"}, format = \hlstr{"BED"})

smooth_sub <- \hlkwd{runmean}(substraction_A7K9_E5K9, 1000, endrule = \hlstr{"constant"})
\hlkwd{export}(smooth_sub, con = \hlstr{"/home/lucas/ISGlobal/Chip_Seq/DATA/Substractions/A7K9_E5K9_substr_smooth1000.bed"}, format = \hlstr{"BED"})

smoothviews <- \hlkwd{lapply}(substraction_A7K9_E5K9, \hlkwd{function}(x) \{\hlkwd{successiveViews}(x, width = 10)\})
\hlkwd{lapply}(smoothviews, \hlkwd{function}(x) \{\hlkwd{mean}(x)\})
\hlcom{#class(substraction_A7K9_E5K9)}
\end{alltt}
\end{kframe}
\end{knitrout}

\section{Session Info}%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{knitrout}
\definecolor{shadecolor}{rgb}{0.969, 0.969, 0.969}\color{fgcolor}\begin{kframe}
\begin{alltt}
\hlkwd{sessionInfo}\hlstd{()}
\end{alltt}
\begin{verbatim}
## R version 3.4.1 (2017-06-30)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: Ubuntu 16.04.3 LTS
## 
## Matrix products: default
## BLAS: /usr/lib/openblas-base/libblas.so.3
## LAPACK: /usr/lib/libopenblasp-r0.2.18.so
## 
## locale:
##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=es_ES.UTF-8        LC_COLLATE=en_US.UTF-8    
##  [5] LC_MONETARY=es_ES.UTF-8    LC_MESSAGES=en_US.UTF-8   
##  [7] LC_PAPER=es_ES.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=es_ES.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
## [1] parallel  stats4    stats     graphics  grDevices utils     datasets 
## [8] methods   base     
## 
## other attached packages:
##  [1] rtracklayer_1.36.4         GenomicAlignments_1.12.2  
##  [3] Rsamtools_1.28.0           Biostrings_2.44.2         
##  [5] XVector_0.16.0             SummarizedExperiment_1.6.3
##  [7] DelayedArray_0.2.7         matrixStats_0.52.2        
##  [9] Biobase_2.36.2             GenomicRanges_1.28.5      
## [11] GenomeInfoDb_1.12.2        IRanges_2.10.3            
## [13] S4Vectors_0.14.4           BiocGenerics_0.22.0       
## [15] knitr_1.17                
## 
## loaded via a namespace (and not attached):
##  [1] magrittr_1.5            zlibbioc_1.22.0        
##  [3] BiocParallel_1.10.1     lattice_0.20-35        
##  [5] highr_0.6               stringr_1.2.0          
##  [7] tools_3.4.1             grid_3.4.1             
##  [9] Matrix_1.2-11           GenomeInfoDbData_0.99.0
## [11] bitops_1.0-6            RCurl_1.95-4.8         
## [13] evaluate_0.10.1         stringi_1.1.5          
## [15] compiler_3.4.1          XML_3.98-1.3
\end{verbatim}
\end{kframe}
\end{knitrout}
\end{document}
