\documentclass{article}\usepackage[]{graphicx}\usepackage[]{color}
%% maxwidth is the original width if it is less than linewidth
%% otherwise use linewidth (to make sure the graphics do not exceed the margin)
\makeatletter
\def\maxwidth{ %
  \ifdim\Gin@nat@width>\linewidth
    \linewidth
  \else
    \Gin@nat@width
  \fi
}
\makeatother

\definecolor{fgcolor}{rgb}{0.345, 0.345, 0.345}
\newcommand{\hlnum}[1]{\textcolor[rgb]{0.686,0.059,0.569}{#1}}%
\newcommand{\hlstr}[1]{\textcolor[rgb]{0.192,0.494,0.8}{#1}}%
\newcommand{\hlcom}[1]{\textcolor[rgb]{0.678,0.584,0.686}{\textit{#1}}}%
\newcommand{\hlopt}[1]{\textcolor[rgb]{0,0,0}{#1}}%
\newcommand{\hlstd}[1]{\textcolor[rgb]{0.345,0.345,0.345}{#1}}%
\newcommand{\hlkwa}[1]{\textcolor[rgb]{0.161,0.373,0.58}{\textbf{#1}}}%
\newcommand{\hlkwb}[1]{\textcolor[rgb]{0.69,0.353,0.396}{#1}}%
\newcommand{\hlkwc}[1]{\textcolor[rgb]{0.333,0.667,0.333}{#1}}%
\newcommand{\hlkwd}[1]{\textcolor[rgb]{0.737,0.353,0.396}{\textbf{#1}}}%
\let\hlipl\hlkwb

\usepackage{framed}
\makeatletter
\newenvironment{kframe}{%
 \def\at@end@of@kframe{}%
 \ifinner\ifhmode%
  \def\at@end@of@kframe{\end{minipage}}%
  \begin{minipage}{\columnwidth}%
 \fi\fi%
 \def\FrameCommand##1{\hskip\@totalleftmargin \hskip-\fboxsep
 \colorbox{shadecolor}{##1}\hskip-\fboxsep
     % There is no \\@totalrightmargin, so:
     \hskip-\linewidth \hskip-\@totalleftmargin \hskip\columnwidth}%
 \MakeFramed {\advance\hsize-\width
   \@totalleftmargin\z@ \linewidth\hsize
   \@setminipage}}%
 {\par\unskip\endMakeFramed%
 \at@end@of@kframe}
\makeatother

\definecolor{shadecolor}{rgb}{.97, .97, .97}
\definecolor{messagecolor}{rgb}{0, 0, 0}
\definecolor{warningcolor}{rgb}{1, 0, 1}
\definecolor{errorcolor}{rgb}{1, 0, 0}
\newenvironment{knitrout}{}{} % an empty environment to be redefined in TeX

\usepackage{alltt}
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
\begin{document}

\begin{knitrout}
\definecolor{shadecolor}{rgb}{0.969, 0.969, 0.969}\color{fgcolor}\begin{kframe}
\begin{alltt}
\hlkwd{source}\hlstd{(}\hlstr{"https://bioconductor.org/biocLite.R"}\hlstd{)}
\end{alltt}


{\ttfamily\noindent\itshape\color{messagecolor}{\#\# Bioconductor version 3.4 (BiocInstaller 1.24.0), ?biocLite for help}}\begin{alltt}
\hlkwd{biocLite}\hlstd{(}\hlstr{"GenomicRanges"}\hlstd{)}
\end{alltt}


{\ttfamily\noindent\itshape\color{messagecolor}{\#\# BioC\_mirror: https://bioconductor.org}}

{\ttfamily\noindent\itshape\color{messagecolor}{\#\# Using Bioconductor 3.4 (BiocInstaller 1.24.0), R 3.3.3 (2017-03-06).}}

{\ttfamily\noindent\itshape\color{messagecolor}{\#\# Installing package(s) 'GenomicRanges'}}

{\ttfamily\noindent\itshape\color{messagecolor}{\#\# installation path not writeable, unable to update packages: XML, cluster,\\\#\#\ \  mgcv, survival}}\begin{alltt}
\hlkwd{biocLite}\hlstd{(}\hlstr{"rtracklayer"}\hlstd{)}
\end{alltt}


{\ttfamily\noindent\itshape\color{messagecolor}{\#\# BioC\_mirror: https://bioconductor.org}}

{\ttfamily\noindent\itshape\color{messagecolor}{\#\# Using Bioconductor 3.4 (BiocInstaller 1.24.0), R 3.3.3 (2017-03-06).}}

{\ttfamily\noindent\itshape\color{messagecolor}{\#\# Installing package(s) 'rtracklayer'}}

{\ttfamily\noindent\itshape\color{messagecolor}{\#\# installation path not writeable, unable to update packages: XML, cluster,\\\#\#\ \  mgcv, survival}}\begin{alltt}
\hlkwd{biocLite}\hlstd{(}\hlstr{"GenomicAlignments"}\hlstd{)}
\end{alltt}


{\ttfamily\noindent\itshape\color{messagecolor}{\#\# BioC\_mirror: https://bioconductor.org}}

{\ttfamily\noindent\itshape\color{messagecolor}{\#\# Using Bioconductor 3.4 (BiocInstaller 1.24.0), R 3.3.3 (2017-03-06).}}

{\ttfamily\noindent\itshape\color{messagecolor}{\#\# Installing package(s) 'GenomicAlignments'}}

{\ttfamily\noindent\itshape\color{messagecolor}{\#\# installation path not writeable, unable to update packages: XML, cluster,\\\#\#\ \  mgcv, survival}}\begin{alltt}
\hlkwd{library}\hlstd{(}\hlstr{"GenomicRanges"}\hlstd{)}
\end{alltt}


{\ttfamily\noindent\itshape\color{messagecolor}{\#\# Loading required package: stats4}}

{\ttfamily\noindent\itshape\color{messagecolor}{\#\# Loading required package: BiocGenerics}}

{\ttfamily\noindent\itshape\color{messagecolor}{\#\# Loading required package: parallel}}

{\ttfamily\noindent\itshape\color{messagecolor}{\#\# \\\#\# Attaching package: 'BiocGenerics'}}

{\ttfamily\noindent\itshape\color{messagecolor}{\#\# The following objects are masked from 'package:parallel':\\\#\# \\\#\#\ \ \ \  clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,\\\#\#\ \ \ \  clusterExport, clusterMap, parApply, parCapply, parLapply,\\\#\#\ \ \ \  parLapplyLB, parRapply, parSapply, parSapplyLB}}

{\ttfamily\noindent\itshape\color{messagecolor}{\#\# The following objects are masked from 'package:stats':\\\#\# \\\#\#\ \ \ \  IQR, mad, xtabs}}

{\ttfamily\noindent\itshape\color{messagecolor}{\#\# The following objects are masked from 'package:base':\\\#\# \\\#\#\ \ \ \  anyDuplicated, append, as.data.frame, cbind, colnames,\\\#\#\ \ \ \  do.call, duplicated, eval, evalq, Filter, Find, get, grep,\\\#\#\ \ \ \  grepl, intersect, is.unsorted, lapply, lengths, Map, mapply,\\\#\#\ \ \ \  match, mget, order, paste, pmax, pmax.int, pmin, pmin.int,\\\#\#\ \ \ \  Position, rank, rbind, Reduce, rownames, sapply, setdiff,\\\#\#\ \ \ \  sort, table, tapply, union, unique, unsplit, which, which.max,\\\#\#\ \ \ \  which.min}}

{\ttfamily\noindent\itshape\color{messagecolor}{\#\# Loading required package: S4Vectors}}

{\ttfamily\noindent\itshape\color{messagecolor}{\#\# \\\#\# Attaching package: 'S4Vectors'}}

{\ttfamily\noindent\itshape\color{messagecolor}{\#\# The following objects are masked from 'package:base':\\\#\# \\\#\#\ \ \ \  colMeans, colSums, expand.grid, rowMeans, rowSums}}

{\ttfamily\noindent\itshape\color{messagecolor}{\#\# Loading required package: IRanges}}

{\ttfamily\noindent\itshape\color{messagecolor}{\#\# Loading required package: GenomeInfoDb}}\begin{alltt}
\hlkwd{library}\hlstd{(}\hlstr{"GenomicAlignments"}\hlstd{)}
\end{alltt}


{\ttfamily\noindent\itshape\color{messagecolor}{\#\# Loading required package: SummarizedExperiment}}

{\ttfamily\noindent\itshape\color{messagecolor}{\#\# Loading required package: Biobase}}

{\ttfamily\noindent\itshape\color{messagecolor}{\#\# Welcome to Bioconductor\\\#\# \\\#\#\ \ \ \  Vignettes contain introductory material; view with\\\#\#\ \ \ \  'browseVignettes()'. To cite Bioconductor, see\\\#\#\ \ \ \  'citation("{}Biobase"{})', and for packages 'citation("{}pkgname"{})'.}}

{\ttfamily\noindent\itshape\color{messagecolor}{\#\# Loading required package: Biostrings}}

{\ttfamily\noindent\itshape\color{messagecolor}{\#\# Loading required package: XVector}}

{\ttfamily\noindent\itshape\color{messagecolor}{\#\# Loading required package: Rsamtools}}\begin{alltt}
\hlkwd{library}\hlstd{(}\hlstr{"rtracklayer"}\hlstd{)}
\hlkwd{library}\hlstd{(parallel)}
\end{alltt}
\end{kframe}
\end{knitrout}


\begin{knitrout}
\definecolor{shadecolor}{rgb}{0.969, 0.969, 0.969}\color{fgcolor}\begin{kframe}
\begin{alltt}
\hlstd{filepaths} \hlkwb{<-} \hlstr{"/home/lucas/ISGlobal/TestSet/align_tests/params_1/"}
\hlstd{files} \hlkwb{<-} \hlkwd{c}\hlstd{(}\hlstr{"/home/lucas/ISGlobal/TestSet/align_tests/params_1/A7K9_14456_TTAGGC_sort.bam"}\hlstd{,} \hlstr{"/home/lucas/ISGlobal/TestSet/align_tests/params_1/E5K9_14457_ACTTGA_sort.bam"}\hlstd{)}
\hlcom{#samples <- files[substrR(files, 8) == "sort.bam"]}

\hlstd{aligns} \hlkwb{<-} \hlkwd{lapply}\hlstd{(files, readGAlignmentPairs)}
\end{alltt}
\end{kframe}
\end{knitrout}

%% ---------------------------------------------------------------------------
\subsection{Calculating Coverage}
\begin{knitrout}
\definecolor{shadecolor}{rgb}{0.969, 0.969, 0.969}\color{fgcolor}\begin{kframe}
\begin{alltt}
\hlstd{cov} \hlkwb{<-} \hlkwd{lapply}\hlstd{(aligns, coverage)}
\end{alltt}
\end{kframe}
\end{knitrout}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section{Normalization}%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%% ----------------------------------------------------------------------------
\subsection{By Number of Mapped Reads} %% -------------------------------------
%% samtools view -F 0x4 foo_sorted.bam | cut -f 1 | sort | uniq | wc -l 

\noindent To get the number of mapped reads in the sample we use \texttt{samtools} as follows:

\begin{verbatim}
samtools view -F 0x4 foo.sorted.bam | cut -f 1 | sort | uniq | wc -l
-F : tells samtools to exclude given flag. Flag 0x4 means unaligned.
\end{verbatim}

\begin{knitrout}
\definecolor{shadecolor}{rgb}{0.969, 0.969, 0.969}\color{fgcolor}\begin{kframe}
\begin{alltt}
\hlstd{nreads} \hlkwb{<-} \hlkwd{c}\hlstd{()}
\hlkwa{for} \hlstd{(file} \hlkwa{in} \hlstd{files) \{}
  \hlstd{cmd} \hlkwb{<-} \hlkwd{paste0}\hlstd{(}\hlstr{"/home/lucas/Programs/samtools-1.3.1/samtools view -F 0x4 "}\hlstd{, file,} \hlstr{" | cut -f 1 | sort | uniq | wc -l"}\hlstd{)}
  \hlstd{nreads} \hlkwb{<-} \hlkwd{c}\hlstd{(nreads,}\hlkwd{system}\hlstd{(cmd,} \hlkwc{intern} \hlstd{=} \hlnum{TRUE}\hlstd{))}
\hlstd{\}}
\end{alltt}
\end{kframe}
\end{knitrout}

\subsection{Normalize Coverage}
\begin{knitrout}
\definecolor{shadecolor}{rgb}{0.969, 0.969, 0.969}\color{fgcolor}\begin{kframe}
\begin{alltt}
\hlstd{norm_cov} \hlkwb{<-} \hlkwd{c}\hlstd{()}
\hlkwa{for} \hlstd{(i} \hlkwa{in} \hlkwd{range}\hlstd{(}\hlnum{1}\hlstd{,}\hlkwd{length}\hlstd{(nreads)))\{}
  \hlstd{norm_cov} \hlkwb{<-} \hlkwd{c}\hlstd{(norm_cov, cov[[i]]}\hlopt{/}\hlkwd{as.numeric}\hlstd{(nreads[i])}\hlopt{*}\hlnum{1000000}\hlstd{)}
\hlstd{\}}
\end{alltt}
\end{kframe}
\end{knitrout}


\begin{knitrout}
\definecolor{shadecolor}{rgb}{0.969, 0.969, 0.969}\color{fgcolor}\begin{kframe}
\begin{alltt}
\hlkwa{for} \hlstd{(i} \hlkwa{in} \hlkwd{range}\hlstd{(}\hlnum{1}\hlstd{,} \hlkwd{length}\hlstd{(norm_cov)))\{}
  \hlstd{con} \hlkwb{<-} \hlkwd{gsub}\hlstd{(}\hlstr{"_sort.bam"}\hlstd{,} \hlstr{"_cov.bed"}\hlstd{, files[i])}
  \hlkwd{export}\hlstd{(norm_cov[[i]],} \hlkwc{con} \hlstd{= con,} \hlkwc{format} \hlstd{=} \hlstr{"BED"}\hlstd{)}
\hlstd{\}}
\end{alltt}
\end{kframe}
\end{knitrout}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section{Substractions}%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{knitrout}
\definecolor{shadecolor}{rgb}{0.969, 0.969, 0.969}\color{fgcolor}\begin{kframe}
\begin{alltt}
\hlstd{substraction_A7K9_E5K9} \hlkwb{<-} \hlstd{norm_cov[[}\hlnum{1}\hlstd{]]} \hlopt{-} \hlstd{norm_cov[[}\hlnum{2}\hlstd{]]}
\hlkwd{export}\hlstd{(substraction_A7K9_E5K9,} \hlkwc{con} \hlstd{=} \hlstr{"/home/lucas/ISGlobal/TestSet/align_tests/params_1/A7K9_E5K9_substr.bed"}\hlstd{,} \hlkwc{format} \hlstd{=} \hlstr{"BED"}\hlstd{)}
\end{alltt}
\end{kframe}
\end{knitrout}

\section{Session Info}%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{knitrout}
\definecolor{shadecolor}{rgb}{0.969, 0.969, 0.969}\color{fgcolor}\begin{kframe}
\begin{alltt}
\hlkwd{sessionInfo}\hlstd{()}
\end{alltt}
\begin{verbatim}
## R version 3.3.3 (2017-03-06)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: Ubuntu 16.04.2 LTS
## 
## locale:
##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=es_ES.UTF-8        LC_COLLATE=en_US.UTF-8    
##  [5] LC_MONETARY=es_ES.UTF-8    LC_MESSAGES=en_US.UTF-8   
##  [7] LC_PAPER=es_ES.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=es_ES.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
## [1] parallel  stats4    stats     graphics  grDevices utils     datasets 
## [8] methods   base     
## 
## other attached packages:
##  [1] rtracklayer_1.34.2         GenomicAlignments_1.10.1  
##  [3] Rsamtools_1.26.1           Biostrings_2.42.1         
##  [5] XVector_0.14.1             SummarizedExperiment_1.4.0
##  [7] Biobase_2.34.0             GenomicRanges_1.26.4      
##  [9] GenomeInfoDb_1.10.3        IRanges_2.8.2             
## [11] S4Vectors_0.12.2           BiocGenerics_0.20.0       
## [13] BiocInstaller_1.24.0       knitr_1.15.1              
## 
## loaded via a namespace (and not attached):
##  [1] magrittr_1.5       zlibbioc_1.20.0    BiocParallel_1.8.1
##  [4] lattice_0.20-34    highr_0.6          stringr_1.2.0     
##  [7] tools_3.3.3        grid_3.3.3         Matrix_1.2-8      
## [10] bitops_1.0-6       RCurl_1.95-4.8     evaluate_0.10     
## [13] stringi_1.1.3      XML_3.98-1.3
\end{verbatim}
\end{kframe}
\end{knitrout}
\end{document}
